test/JDBC/expected/GRANT_SCHEMA.out

```diff
@@ -0,0 +1,845 @@
+go
+
+select * from αγάπη.abc;
+go
+~~START~~
+int
+~~END~~
+
+
+select * from babel_4344_S1.babel_4344_t1;
+go
+~~START~~
+int
+~~END~~
+
+
+use master;
+go
+
+-- tsql user=babel_4344_l1 password=12345678 
+use babel_4344_d1;
+go
+
+-- User has select privileges, tables and views be accessible
+select * from babel_4344_s1.babel_4344_t1
+go
+~~START~~
+int
+~~END~~
+
+select * from babel_4344_s1.babel_4344_v1;
+go
+~~START~~
+int
+2
+~~END~~
+
+use master;
+go
+
+-- tsql
+use babel_4344_d1;
+go
+revoke select on schema::babel_4344_s1 from public, αιώνια;
+go
+
+-- tsql user=babel_4344_l1 password=12345678 
+use babel_4344_d1;
+go
+
+-- User doesn't have any privileges, objects should not be accessible
+select * from babel_4344_t1;
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: permission denied for table babel_4344_t1)~~
+
+select * from babel_4344_s1.babel_4344_t1
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: permission denied for table babel_4344_t1)~~
+
+insert into babel_4344_s1.babel_4344_t1 values(1);
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: permission denied for table babel_4344_t1)~~
+
+select * from babel_4344_v1;
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: permission denied for view babel_4344_v1)~~
+
+select * from babel_4344_s1.babel_4344_v1;
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: permission denied for view babel_4344_v1)~~
+
+exec babel_4344_p1;
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: permission denied for procedure babel_4344_p1)~~
+
+exec babel_4344_s1.babel_4344_p1;
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: permission denied for procedure babel_4344_p1)~~
+
+select * from babel_4344_f1();
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: permission denied for function babel_4344_f1)~~
+
+select * from babel_4344_s1.babel_4344_f1();
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: permission denied for function babel_4344_f1)~~
+
+use master;
+go
+
+-- tsql
+-- GRANT OBJECT privilege
+use babel_4344_d1;
+go
+grant SELECT on babel_4344_t1 to BABEL_4344_U1;
+go
+grant SELECT on babel_4344_s1.babel_4344_t1 to babel_4344_u1;
+go
+grant all on babel_4344_s1.babel_4344_t1 to babel_4344_u1;
+go
+grant select on babel_4344_t3(a) to babel_4344_u1; -- column privilege
+go
+grant select on babel_4344_s1.babel_4344_t3(a) to babel_4344_u1; -- column privilege
+go
+grant select on babel_4344_v1 to babel_4344_u1;
+go
+grant select on babel_4344_s1.babel_4344_v1 to babel_4344_u1;
+go
+grant execute on babel_4344_p1 to babel_4344_u1;
+go
+grant execute on babel_4344_s1.babel_4344_p1 to babel_4344_u1;
+go
+-- inside a transaction, permission will not be granted since it is rolled back
+begin transaction;
+exec sp_executesql N'grant execute on babel_4344_s1.babel_4344_p3 to babel_4344_u1;';
+rollback transaction;
+go
+
+-- Mixed case
+grant Execute on Babel_4344_F1 to babel_4344_u1;
+go
+grant execute on babel_4344_s1.babel_4344_f1 to babel_4344_u1;
+go
```

Comment:
user should also be in mixed cases. Where/how are we verifying if this works as expected?

Reply:
schema name should also be in mixed cases

Reply:
grant select on schema in one case and then use different case during revoke or different case during select. Same should be done at object level

Reply:
All these tests are added. Where the grant on schema/object is in one case and revoke/select/execute is in a different case.